### ISSUE TITLE
Shape inference time doubled after #649

### ISSUE THREAD

**Issue by zasdfgbnm** [NVIDIA/Fuser#749]
Before #649:
```C++
-------------------------------------------------------------------------------------------
Benchmark                                                 Time             CPU   Iterations
-------------------------------------------------------------------------------------------
LayerNormBackward_ShapeInference                       87.8 us         87.6 us         7758
LayerNormForward_ShapeInference                        29.9 us         29.8 us        22938
LayerNormBackward_NoShapeInferenceCachedBaseline       36.5 us         36.5 us        18906
LayerNormForward_NoShapeInferenceCachedBaseline        21.4 us         21.3 us        33063
```
After:
```C++
-------------------------------------------------------------------------------------------
Benchmark                                                 Time             CPU   Iterations
-------------------------------------------------------------------------------------------
LayerNormBackward_ShapeInference                        162 us          162 us         4332
LayerNormForward_ShapeInference                        50.3 us         50.2 us        13811
LayerNormBackward_NoShapeInferenceCachedBaseline        111 us          111 us         6304
LayerNormForward_NoShapeInferenceCachedBaseline        42.4 us         42.3 us        16632
```

**Comment by naoyam** [comment]
Any plan for this issue?

**Comment by IvanYashchuk** [comment]
Here are the results with https://github.com/NVIDIA/Fuser/pull/768
```
[------------  -----------]
           |  With #768 PR
1 threads: ----------------
      2    |       25.3
      4    |       34.6
      8    |       52.6
      16   |       86.6
      32   |      179.8
      64   |      393.0
      128  |      743.2

Times are in microseconds (us).
```

**Comment by zasdfgbnm** [comment]
With https://github.com/NVIDIA/Fuser/pull/813, the cpu overhead is almost back. Closing this as resolved. Feel free to reopen or open a new issue if we need to optimize cpu overhead further.
